gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2291,6 +2291,21 @@ func findGotoLoopDefines(ctx BlockNode, bn BlockNode) {
 			}()
 
 			switch n := n.(type) {
+			case *ForStmt:
+				Transcribe(n,
+					func(ns []Node, ftype TransField, index int, n Node, stage TransStage) (Node, TransCtrl) {
+						switch stage {
+						case TRANS_ENTER:
+							switch n := n.(type) {
+							case *NameExpr:
+								if n.Type == NameExprTypeDefine {
+									n.Type = NameExprTypeHeapDefine
+								}
+							}
+						}
+						return n, TRANS_CONTINUE
+					})
+
```

Comment:
this is missing a lot of the logic that is needed, which is present in the BranchStmt's Transcribe.
TRANS_ENTER *FuncLitExpr should return SKIP similar, because LoopDefines should be defined within the for-loop but not within a containing closure.

Reply:
fixed.